.github/workflows/build.yml

```diff
@@ -81,6 +81,112 @@ jobs:
       - run: cd clients/js && make install
       - run: cd ethereum && make test-upgrade
 
+  solana:
+    runs-on: ubuntu-20.04
+    env:
+      RUSTFLAGS: -Dwarnings
+      EMITTER_ADDRESS: CiByUvEcx7w2HA4VHcPCBUAFQ73Won9kB36zW9VjirSr
+      BRIDGE_ADDRESS: Bridge1p5gheXUvJ6jGWGeCsgPKgnE3YgdGKRVCMY9o
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get rust toolchain version
+        id: toolchain
+        run: |
+          RUST_VERSION="$(awk '/channel =/ { print substr($3, 2, length($3)-2) }' solana/rust-toolchain)"
+          echo "::set-output name=version::${RUST_VERSION}"
+
+      - name: Get solana version
+        id: solana
+        run: |
+          SOLANA_VERSION="$(awk '/solana-program =/ { print substr($3, 3, length($3)-3) }' solana/bridge/program/Cargo.toml)"
+          echo "::set-output name=version::${SOLANA_VERSION}"
+
+      - name: Cache rust toolchain
+        uses: actions/cache@v3
+        env:
+          cache-name: solana-toolchain
+        with:
+          path: |
+            ~/.cargo/bin
+            ~/.rustup
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ steps.toolchain.outputs.version }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+            toolchain: ${{ steps.toolchain.outputs.version }}
+            components: "clippy,rustfmt"
+
+      - name: Cache rust packages
+        uses: actions/cache@v3
+        env:
+          cache-name: solana-rust-packages
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('solana/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Run `cargo fmt`
+        run: cargo fmt --check --all --manifest-path solana/Cargo.toml
```

Comment:
how about adding these commands as make targets so they can more conveniently be invoked locally too?

Reply:
Done.